File: ClientApplication/OnlineClothingStore/src/app/models/order.ts
import { OrderItem } from './orderItem';
import { Status } from './status';
import { User } from './user';

export type Order = {
  id: number;
  orderdate: Date;
  statusId: number;
  userId: number;

  status: Status;
  user?: User;

  orderItems: OrderItem[];
};

File: Models/Order.cs
using System;
using System.Collections.Generic;

namespace OnlineClothingStore.Models;

public partial class Order
{
    public int Id { get; set; }

    public DateOnly OrderDate { get; set; }

    public int StatusId { get; set; }

    public int UserId { get; set; }

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();

    public virtual Status Status { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
